x-pack/plugins/security_solution/public/resolver/store/middleware/current_related_event_fetcher.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Dispatch, MiddlewareAPI } from 'redux';
+import { isEqual } from 'lodash';
+import { SafeResolverEvent } from '../../../../common/endpoint/types';
+
+import { ResolverState, DataAccessLayer, PanelViewAndParameters } from '../../types';
+import * as selectors from '../selectors';
+import { ResolverAction } from '../actions';
+
+/**
+ *
+ * @description - This api is called after every state change.
+ * If the current view is the `eventDetail` view it will request the event details from the server.
+ * @export
+ * @param {DataAccessLayer} dataAccessLayer
+ * @param {MiddlewareAPI<Dispatch<ResolverAction>, ResolverState>} api
+ * @returns {() => void}
+ */
+export function CurrentRelatedEventFetcher(
+  dataAccessLayer: DataAccessLayer,
+  api: MiddlewareAPI<Dispatch<ResolverAction>, ResolverState>
+): () => void {
+  let last: PanelViewAndParameters | undefined;
+
+  return async () => {
+    const state = api.getState();
+
+    const newParams = selectors.panelViewAndParameters(state);
+
+    const oldParams = last;
+    last = newParams;
+
+    // If the panel view params have changed and the current panel view is the `eventDetail`, then fetch the event details for that eventID.
+    if (!isEqual(newParams, oldParams) && newParams.panelView === 'eventDetail') {
+      const currentEventID = newParams.panelParameters.eventID;
+
+      api.dispatch({
+        type: 'appRequestedCurrentRelatedEventData',
+      });
+
+      let result: SafeResolverEvent | undefined | null;
```

Comment:
Do we need `undefined` and `null`. I saw that the state starts out as `null` in the DataState, could we get away with just using `undefined` though?

Reply:
Yea, I thought about that, it'll probably be easier/cleaner that way. I typically try not to explicitly set things to undefined if it isn't 100% necessary, as I like to leave `undefined` as the representation for variables that were never created/initialized/ etc... Not a hard and fast rule, but I try to stick to it where I can.